app/scripts/modules/core/src/application/ApplicationPostSearchResultSearcher.ts

```diff
@@ -0,0 +1,45 @@
+import { IPromise } from 'angular';
+
+import { ApplicationReader, IApplicationSummary } from './service/application.read.service';
+import { IPostSearchResultSearcher } from 'core/search/searchResult/PostSearchResultSearcherRegistry';
+import { ISearchResult } from 'core/search/search.service';
+import { ISearchResultFormatter, searchResultFormatterRegistry } from 'core/search/searchResult/searchResultFormatter.registry';
+import { ISearchResultSet } from 'core/search/infrastructure/infrastructureSearch.service';
+import { IServerGroupSearchResult } from 'core/search/searchResult/model/IServerGroupSearchResult';
+
+export class ApplicationPostSearchResultSearcher implements IPostSearchResultSearcher<IServerGroupSearchResult> {
+
+  private static TYPE = 'applications';
+
+  constructor(private applicationReader: ApplicationReader) {}
+
+  public getPostSearchResults(inputs: IServerGroupSearchResult[]): IPromise<ISearchResultSet[]> {
+
+    const names: Set<string> = new Set<string>(inputs.map((result: IServerGroupSearchResult) => result.application).sort());
+    return this.applicationReader.listApplications().then((apps: IApplicationSummary[]) => {
+
+      const results: ISearchResult[] = apps.filter((app: IApplicationSummary) => names.has(app.name))
+        .map((app: IApplicationSummary) => {
+          return {
+            accounts: app.accounts,
+            application: app.name,
+            displayName: app.name,
+            href: `#/${ApplicationPostSearchResultSearcher.TYPE}/${app.name}/clusters`,
```

Comment:
any reason not to use `urlBuilder` here?

Reply:
Doesn't look like it.  I'd have to add `$state` to the ctor, but that's not really a big deal...